jwst/stpipe/step.py

```diff
@@ -1093,12 +1092,12 @@ def closeout(self, to_close=None, to_del=None):
                 self.log.debug("An error has occurred: %s", error)
         gc.collect()
 
-    @abc.abstractclassmethod
+    @classmethod
     def datamodels_open(cls, init, **kwargs):
         """
         Wrapper around observatory-specific datamodels.open function.
         """
-        pass
+        raise NotImplementedError(f"{cls.__name__} does not implement datamodels_open")
```

Comment:
What is the purpose of `datamodels_open` class method, given that we already have `open_model`?  I see it is called by `open_model` method below, but is this supposed to be part of the top-level user interface for `Step`?

I guess I'm asking if it should be a private method instead?

Reply:
`datamodels_open` is a class method, which the CLI code needs in order to open the dataset and request step parameters from CRDS.  I think it would be reasonable to make it private, at least until a user demonstrates a clear need, so I'll go ahead and do that.